### PR TITLE
DOMA 2308 make `sender` and `dv` fields required. Use `dvAndSender` plugin

### PR DESCRIPTION
**Pull request by VKislov**
Transferring the addition of `sender` and `dv` fields to the plugin `dvAndSender`

### CODE REVIEW

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js [hunk 1]
```diff
@@ -191,17 +193,27 @@ describe('AcquiringIntegration', () => {
...
-            }, DV_UNKNOWN_VERSION_ERROR)
+            await expectToThrowGQLError(async () => await createTestAcquiringIntegration(admin, [billingIntegration], {
+                dv: 2,
+            }), {
+                'code': 'BAD_USER_INPUT',
+                'type': 'DV_VERSION_MISMATCH',
+                'message': 'Wrong value for data version number',
+                'mutation': 'createAcquiringIntegration',
+                'messageForUser': '',
+                'variable': ['data', 'dv'],
```
Comment (toplenboren) [inline apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js:204]: i think this is enough to check on error type

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js [hunk 2]
```diff
@@ -191,17 +193,27 @@ describe('AcquiringIntegration', () => {
...
-            }, DV_UNKNOWN_VERSION_ERROR)
+            await expectToThrowGQLError(async () => await updateTestAcquiringIntegration(admin, integration.id, {
+                dv: 2,
+            }), {
+                'code': 'BAD_USER_INPUT',
+                'type': 'DV_VERSION_MISMATCH',
+                'message': 'Wrong value for data version number',
+                'mutation': 'updateAcquiringIntegration',
+                'messageForUser': '',
+                'variable': ['data', 'dv'],
```
Comment (toplenboren) [inline apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js:215]: ^^ and so on ><

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
 import { DV_UNKNOWN_VERSION_ERROR } from '@condo/domains/common/constants/errors'
 import { NO_INSTRUCTION_OR_MESSAGE_ERROR } from '@condo/domains/miniapp/constants'
+import { expectToThrowGQLError } from '../../common/utils/testSchema'
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js:7]:
```suggestion
import { expectToThrowGQLError } from '@condo/domains/common/utils/testSchema'
```

File: apps/condo/domains/acquiring/schema/MultiPayment.test.js [hunk 4]
```diff
@@ -67,6 +67,9 @@ const dayjs = require('dayjs')
 const faker = require('faker')
 const omit = require('lodash/omit')
 const get = require('lodash/get')
+const { expectToThrowGQLError } = require('../../common/utils/testSchema')
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.test.js:70]:
```suggestion
const { expectToThrowGQLError } = require('@condo/domains/common/utils/testSchema')
```

File: apps/condo/domains/acquiring/schema/MultiPayment.test.js [hunk 5]
```diff
@@ -706,23 +734,31 @@ describe('MultiPayment', () => {
                 const explicitServiceCharge = '0'
                 let [payment] = payments
                 await MultiPayment.update(integrationClient, multiPaymentId, {
+                    dv: 1,
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.test.js:737]: use `updateTestMultiPayment`

File: apps/condo/domains/acquiring/schema/MultiPayment.test.js [hunk 6]
```diff
@@ -706,23 +734,31 @@ describe('MultiPayment', () => {
...
+                    sender,
                     status: MULTIPAYMENT_PROCESSING_STATUS,
                     explicitFee,
                     explicitServiceCharge,
                 })
                 await Payment.update(integrationClient, payment.id, {
+                    dv: 1,
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.test.js:744]: use `updateTestPayment`

File: apps/condo/domains/acquiring/schema/MultiPayment.test.js [hunk 7]
```diff
@@ -706,23 +734,31 @@ describe('MultiPayment', () => {
...
+                    sender,
                     status: PAYMENT_PROCESSING_STATUS,
                     explicitFee,
                     explicitServiceCharge,
                 })
 
                 // Stage 2. Our acquiring integration service after polling UPS service move everything to ERROR
                 payment = await Payment.update(integrationClient, payment.id, {
+                    dv: 1,
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.test.js:753]: updateTestPayment

File: apps/condo/domains/acquiring/schema/MultiPayment.test.js [hunk 8]
```diff
@@ -706,23 +734,31 @@ describe('MultiPayment', () => {
...
+                    sender,
                     status: PAYMENT_ERROR_STATUS,
                 })
                 expect(payment).toBeDefined()
                 expect(payment).toHaveProperty('status', PAYMENT_ERROR_STATUS)
                 const multiPayment = await MultiPayment.update(integrationClient, multiPaymentId, {
+                    dv: 1,
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/MultiPayment.test.js:760]: updateTestMultiPayment

File: apps/condo/domains/acquiring/schema/Payment.test.js [hunk 9]
```diff
@@ -55,6 +55,8 @@ const {
 } = require('@condo/domains/acquiring/constants/errors')
 const dayjs = require('dayjs')
 const Big = require('big.js')
+const { expectToThrowGQLError } = require('../../common/utils/testSchema')
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/Payment.test.js:58]: !
```suggestion
const { expectToThrowGQLError } = require('@condo/domains/common/utils/testSchema')
```

File: apps/condo/domains/acquiring/schema/PaymentsFilterTemplate.js [hunk 10]
```diff
@@ -5,16 +5,13 @@
 const { Text, Relationship } = require('@keystonejs/fields')
 const { GQLListSchema } = require('@core/keystone/schema')
 const { historical, versioned, uuided, tracked, softDeleted } = require('@core/keystone/plugins')
-const { SENDER_FIELD, DV_FIELD } = require('@condo/domains/common/schema/fields')
 const access = require('@condo/domains/acquiring/access/PaymentsFilterTemplate')
 const { PAYMENTS_FILTER_FIELD } = require('@condo/domains/acquiring/schema/fields/PaymentsFilter')
+const { dvAndSender } = require('../../common/schema/plugins/dvAndSender')
```
Comment (pahaz) [inline apps/condo/domains/acquiring/schema/PaymentsFilterTemplate.js:10]:
```suggestion
const { dvAndSender } = require('@condo/domains/common/schema/plugins/dvAndSender')
```

File: apps/condo/domains/billing/schema/BillingAccount.js [hunk 11]
```diff
@@ -6,24 +6,22 @@ const { Text } = require('@keystonejs/fields')
...
 const {
-    DV_UNKNOWN_VERSION_ERROR,
     JSON_EXPECT_OBJECT_ERROR,
     JSON_UNKNOWN_VERSION_ERROR,
     UNEQUAL_CONTEXT_ERROR,
 } = require('@condo/domains/common/constants/errors')
 const { hasValidJsonStructure, hasDvAndSenderFields } = require('@condo/domains/common/utils/validation.utils')
 const { RAW_DATA_FIELD } = require('./fields/common')
 const { INTEGRATION_CONTEXT_FIELD, BILLING_PROPERTY_FIELD } = require('./fields/relations')
+const { dvAndSender } = require('../../common/schema/plugins/dvAndSender')
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/BillingAccount.js:19]: ! wront import

File: apps/condo/domains/billing/schema/BillingCategory.js [hunk 12]
```diff
@@ -4,16 +4,14 @@
 
 const { GQLListSchema } = require('@core/keystone/schema')
 const { historical, versioned, uuided, tracked, softDeleted } = require('@core/keystone/plugins')
-const { SENDER_FIELD, DV_FIELD } = require('@condo/domains/common/schema/fields')
 const access = require('@condo/domains/billing/access/BillingCategory')
 const { LocalizedText } = require('@core/keystone/fields')
+const { dvAndSender } = require('../../common/schema/plugins/dvAndSender')
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/BillingCategory.js:9]: !

File: apps/condo/domains/billing/schema/BillingIntegration.js [hunk 13]
```diff
@@ -22,19 +21,15 @@ const {
     CONTEXT_DEFAULT_STATUS_FIELD,
 } = require('@condo/domains/miniapp/schema/fields/integration')
 const { ABOUT_DOCUMENT_FIELD } = require('@condo/domains/miniapp/schema/fields/aboutDocumentField')
-const { hasDvAndSenderFields } = require('@condo/domains/common/utils/validation.utils')
 const { NO_INSTRUCTION_OR_MESSAGE_ERROR } = require('@condo/domains/miniapp/constants')
-const { DV_UNKNOWN_VERSION_ERROR } = require('@condo/domains/common/constants/errors')
 const { getFileMetaAfterChange } = require('@condo/domains/common/utils/fileAdapter')
+const { dvAndSender } = require('../../common/schema/plugins/dvAndSender')
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/BillingIntegration.js:26]: !

File: apps/condo/domains/meter/schema/Meter.js [hunk 14]
```diff
@@ -7,24 +7,21 @@ const { Text, Relationship, Integer, DateTimeUtc, Checkbox } = require('@keyston
...
 const access = require('@condo/domains/meter/access/Meter')
-const { DV_UNKNOWN_VERSION_ERROR } = require('@condo/domains/common/constants/errors')
-const { hasDvAndSenderFields } = require('@condo/domains/common/utils/validation.utils')
 const { UNIQUE_ALREADY_EXISTS_ERROR } = require('@condo/domains/common/constants/errors')
 const { RESIDENT } = require('@condo/domains/user/constants/common')
 const { AUTOMATIC_METER_NO_MASTER_APP, B2B_APP_NOT_CONNECTED, B2C_APP_NOT_AVAILABLE } = require('@condo/domains/meter/constants/errors')
 
 const { Meter: MeterApi } = require('./../utils/serverSchema')
 const { MeterReading } = require('../utils/serverSchema')
+const { dvAndSender } = require('../../common/schema/plugins/dvAndSender')
```
Comment (pahaz) [inline apps/condo/domains/meter/schema/Meter.js:19]: !

File: apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js (hunk 2 above)
Comment (toplenboren) [reply apps/condo/domains/acquiring/schema/AcquiringIntegration.test.js:215]: about test

**Review by toplenboren — APPROVED** [review]

File: apps/condo/domains/user/schema/ChangePhoneNumberResidentUserService.js [hunk 15]
```diff
@@ -7,12 +7,29 @@ const access = require('@condo/domains/user/access/ChangePhoneNumberResidentUser
...
 const { NOT_FOUND, DV_VERSION_MISMATCH } = require('@condo/domains/common/constants/errors')
+const { WRONG_FORMAT } = require('@condo/domains/common/constants/errors')
+const { checkDvSender } = require('../../common/utils/serverSchema/validators')
 
 /**
  * List of possible errors, that this custom schema can throw
  * They will be rendered in documentation section in GraphiQL for this custom schema
  */
 const errors = {
+    DV_VERSION_MISMATCH: {
```
Comment (dkoviazin) [inline apps/condo/domains/user/schema/ChangePhoneNumberResidentUserService.js:18]: Looks like a common code that can be moved to utils

File: apps/condo/domains/user/schema/ConfirmPhoneActionService.js [hunk 16]
```diff
@@ -31,6 +34,19 @@ const {
 const redisGuard = new RedisGuard()
 
 const errors = {
+    DV_VERSION_MISMATCH: {
```
Comment (dkoviazin) [inline apps/condo/domains/user/schema/ConfirmPhoneActionService.js:37]: common code = move to utils

**Review by dkoviazin — APPROVED** [review]

**Pull request merged**